src/sfctl/entry.py

```diff
@@ -23,11 +25,46 @@ def cli():
                         help_cls=SFCommandHelp)
 
 
+def check_and_send_telemetry(args_list, invocation_ret_val):
+    """
+    Check if telemetry should be sent, and if so, send the telemetry
+
+    :param args_list: a list of string representing the command called along
+        with its parameters
+    :param invocation_ret_val: The return value of calling invoke on the command in args_list
+
+    :return: None
+    """
+
+    if get_telemetry_config():
+
+        try:
+            # In the background, do some work on checking and sending telemetry for the current call.
+            send_telemetry(args_list, invocation_ret_val)
+        except Exception as e:  # pylint: disable=broad-except
+            # Allow telemetry to fail silently.
+            # TODO: Uncomment out line below to trigger notifying on failure.
+            # print('Telemetry has failed with message "{}".'.format(str(e)))
```

Comment:
Is the plan to display info for when a telemetry event fails?

Reply:
Telemetry failures should be silent, and will be logged to debug channels only.